src/commands/run.rs

```diff
@@ -360,25 +367,42 @@ fn populate_with_wasi(
         builder = builder.preopened_dir(dir, name)?;
     }
 
-    Wasi::new(linker.store(), builder.build()?).add_to_linker(linker)?;
+    if wasi_modules.wasi_common {
+        Wasi::new(linker.store(), builder.build()?).add_to_linker(linker)?;
+    }
 
+    #[cfg(not(feature = "wasi-nn"))]
+    assert!(
+        !wasi_modules.wasi_nn,
+        "Cannot enable wasi-nn when the binary is not compiled with this feature."
+    );
```

Comment:
Instead of being an `assert` could this return a more first-class error to avoid panicking?

Reply:
Ok, I went with `bail!`? I think some time down the road this type of check should be able to go away if we were to remove the special build flags for these WASI modules.